tensorboard/plugins/mesh_visualizer/tf_mesh_dashboard/mesh-loader.js

```diff
@@ -0,0 +1,287 @@
+    // sometimes)
+    this._dataProvider = new vz_mesh.ArrayBufferDataProvider(
+        this.requestManager);
+    this._meshViewer = new vz_mesh.MeshViewer();
+    this._meshViewer.addEventListener(
+        'beforeUpdateScene', this._updateCanvasSize.bind(this));
+    this._meshViewer.addEventListener(
+        'cameraPositionChange', this._onCameraPositionChange.bind(this));
+    this.isAttached = true;
+    this.reload();
+  },
+
+  /**
+   * Function to call when component must be reloaded.
+   */
+  reload: function() {
+    if (!this.isAttached) {
+      return;
+    }
+    this.set('_isMeshLoading', true);
+    this._dataProvider.reload(this.run, this.tag, this.sample).then((steps) => {
+      if (!steps) return;  // Happens when request was cancelled at some point.
+      this.set('_steps', steps);
+      this.set('_stepIndex', steps.length - 1);
+      this.set('_isMeshLoading', false);
+      if (!this._meshViewerAttached) {
+        // Mesh viewer should be added to the dom once.
+        this.appendChild(this._meshViewer.getRenderer().domElement);
+        this._meshViewerAttached = true;
+      }
+    });
+  },
+
+  /**
+   * Updates the scene.
+   * @param {!Object} currentStep Step datum.
+   * @private
+   */
+  _updateScene: function(currentStep) {
+    this._meshViewer.updateScene(currentStep, this);
+    if (!this._cameraPositionInitialized) {
+      this._meshViewer.resetView();
+      this._cameraPositionInitialized = true;
+    }
+  },
+
+  /**
+   * Propagates camera position change event outside of the loader.
+   * @private
+   */
+  _onCameraPositionChange: function() {
+    if (!this._meshViewer.isReady()) return;
+    const event = new CustomEvent('camera-position-change', {
+                              detail: this._meshViewer.getCameraPosition()
+                            });
+    this.dispatchEvent(event);
+  },
+
+  /**
+   * Creates mesh geometry for current step data.
+   * @param {!THREE.Vector3} position Position of the camera.
+   * @param {number} far Camera frustum far plane.
+   * @param {!THREE.Vector3} target Point in space for camera to look at.
+   * @public
+   */
+  setCameraViewpoint: function(position, far, target) {
+    this._meshViewer.setCameraViewpoint(position, far, target);
+  },
+
+  /**
+   * Updates size of the canvas.
+   * @private
+   */
+  _updateCanvasSize: function() {
+    const width = this.offsetWidth;
+    const height = width;  // Keep the whole mesh viewer square at all times.
+    const headerHeight = this.$$(
+        '.tf-mesh-loader-header').offsetHeight;
+    const canvasSize = {
+      width: width,
+      height: height - headerHeight,
+    };
+    this._meshViewer.setCanvasSize(canvasSize);
+  },
+
+  /**
+   * Re-renders component in the browser.
+   * @public
+   */
+  redraw: function() {
+    this._updateCanvasSize();
+    // Do not render if not in the DOM.
+    if (!this.parentElement) return;
```

Comment:
What does this mean? You may want to use `node.isConnected` 
https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected